plugin-server/tests/main/process-event.test.ts

```diff
@@ -1208,6 +1210,79 @@ test('snapshot event stored as session_recording_event', () => {
         window_id: undefined,
     })
 })
+;[
+    {
+        snapshotData: { timestamp: 123 },
+        expected: { click_count: 0, keypress_count: 0, mouse_activity_count: 0, url: undefined },
+    },
+    {
+        snapshotData: { timestamp: 123, events_summary: [{ type: 3, data: { source: 2 } }] },
+        expected: { click_count: 1, keypress_count: 0, mouse_activity_count: 1, url: undefined },
+    },
+    {
+        snapshotData: { timestamp: 123, events_summary: [{ type: 3, data: { source: 5 } }] },
+        expected: { click_count: 0, keypress_count: 1, mouse_activity_count: 1, url: undefined },
+    },
+    {
+        snapshotData: {
+            timestamp: 123,
+            events_summary: [
+                {
+                    timestamp: 1682449093693,
+                    type: 5,
+                    data: {
+                        tag: '$pageview',
+                        payload: {
+                            // doesn't match because href is nested in payload
+                            href: 'http://127.0.0.1:8000/home',
+                        },
+                    },
+                },
+                {
+                    timestamp: 1682449093469,
+                    type: 4,
+                    data: {
+                        href: 'http://127.0.0.1:8000/second/url',
+                        width: 1512,
+                        height: 833,
+                    },
+                },
+            ],
+        },
+        expected: {
+            click_count: 0,
+            keypress_count: 0,
+            mouse_activity_count: 0,
+            url: 'http://127.0.0.1:8000/second/url',
+        },
+    },
+].forEach(({ snapshotData, expected }) => {
+    test(`snapshot event ${JSON.stringify(snapshotData)} can be stored as session_replay_event`, () => {
+        const data = createSessionReplayEvent(
+            'some-id',
+            team.id,
+            '5AzhubH8uMghFHxXq0phfs14JOjH6SA2Ftr1dzXj7U4',
+            now,
+            '',
+            {
+                $session_id: 'abcf-efg',
+                $snapshot_data: snapshotData,
+            } as any as Properties
+        )
+
+        const expectedEvent: SummarizedSessionRecordingEvent = {
+            created_at: expect.stringMatching(/^\d{4}-\d{2}-\d{2} [\d\s:]+/),
+            distinct_id: '5AzhubH8uMghFHxXq0phfs14JOjH6SA2Ftr1dzXj7U4',
+            session_id: 'abcf-efg',
+            team_id: 2,
+            timestamp: expect.stringMatching(/^\d{4}-\d{2}-\d{2} [\d\s:]+/),
```

Comment:
In relation to the comments above we probably want to always test the timestamp is correct, in particular the `first` and `last` timestamps.

Reply:
Am testing the timestamp aggregation on the python/clickhouse side

(i think) we have

```mermaid
flowchart TB
received_snapshot_event --> summarised_event
summarised_event --> kafka
kafka --> clikchouse_kafka_table
clikchouse_kafka_table --mv pulls from kafka\nin a batch\nand groups events--> clickhouse_materialized_view
clickhouse_materialized_view --mv writes interim aggregates to table--> clickhouse_aggregating_table
clickhouse_aggregating_table --query must group\nin order to\nuse data-->query

```

Reply:
Point I was making here was more that once we have the first and last timestamps, we should test explicitly for them here to make sure the logic is correct